### PR TITLE
Implement infectivity radius on Stiefel

### PR DESCRIPTION
**Pull request by kellertuer**
cf. Zimmermann & Stoye https://arxiv.org/abs/2405.02268v1 .

### CODE REVIEW

**Comment by mateuszbaran** [comment]
That was quick! The failure is caused by a new ambiguity:
```
injectivity_radius(M::AbstractDecoratorManifold, m::AbstractRetractionMethod) @ ManifoldsBase ~/.julia/packages/ManifoldsBase/pOL51/src/nested_trait.jl:305,
injectivity_radius(::Stiefel, p) @ Manifolds ~/work/Manifolds.jl/Manifolds.jl/src/manifolds/StiefelEuclideanMetric.jl:112
```

**Comment by kellertuer** [comment]
Oh we should maybe check ambiguities on the infectivity radius.

**Comment by mateuszbaran** [comment]
It's real, you can try calling `injectivity_radius(Stiefel(5, 2), ExponentialRetraction())` and it would fail. You can probably just delete `injectivity_radius(::Stiefel, p) = π` though.

**Comment by kellertuer** [comment]
Then the special point tests fail. But if that is ok, we can do that

File: src/manifolds/StiefelEuclideanMetric.jl [hunk 1]
```diff
@@ -102,7 +102,16 @@ end
...
 
+@doc raw"""
+    injectivity_radius(M::Stiefel[, p])
+
+Return the injectivity radius for the [`Stiefel`](@ref) manifold `M`,
+which is globally ``π`` [ZimmermannStoye:2024](@cite).
 """
+injectivity_radius(::Stiefel) = π
+
+
```
[the remark this answers is not in the document]
Comment (kellertuer) [reply src/manifolds/StiefelEuclideanMetric.jl]: Today is my “I yet again forgot an empty line somewhere” day.

**Comment by mateuszbaran** [comment]
> Then the special point tests fail. But if that is ok, we can do that (that is remove the point ones).

You can limit the type of `p` in `injectivity_radius(::Stiefel, p)` to those special point types.

**Comment by kellertuer** [comment]
or leave them out, we would specialise to arrays or such, and that might not be to useful (we otherwise do not do that I think).

**Comment by mateuszbaran** [comment]
I think the variant with `p` should work. You can also disambiguate by defining `injectivity_radius(M::Stiefel, m::AbstractRetractionMethod)`. Or we shouldn't have made `injectivity_radius` so prone to ambiguities.

**Comment by kellertuer** [comment]
I omitted the p variants for now, but sure in the long run we should make the infectivity radius a bit less prone to ambiguities.

File: src/manifolds/StiefelEuclideanMetric.jl [hunk 2]
```diff
@@ -102,7 +102,15 @@ end
 
 _euclidean_unit_vector(n, i) = [k == i ? 1.0 : 0.0 for k in 1:n]
 
+@doc raw"""
+    injectivity_radius(M::Stiefel[, p])
```
Comment (mateuszbaran) [inline src/manifolds/StiefelEuclideanMetric.jl:106]:
```suggestion
    injectivity_radius(M::Stiefel)
```

**Review by mateuszbaran — APPROVED** [review]

**Pull request merged**